lib/src/main/kotlin/starknet/service/http/HttpRequest.kt

```diff
@@ -25,4 +25,37 @@ class HttpRequest<T>(
             Json.decodeFromString(deserializer, response)
         }
     }
+
+    companion object {
+        fun <T> makeRequest(
+            url: String,
+            method: String,
+            body: String,
+            deserializer: DeserializationStrategy<T>
+        ): HttpRequest<T> {
+            val httpPayload = HttpService.Payload(url, method, body)
+            return HttpRequest(httpPayload, deserializer)
+        }
+
+        fun <T> makeRequest(
+            url: String,
+            method: String,
+            params: List<Pair<String, String>>,
+            deserializer: DeserializationStrategy<T>
+        ): HttpRequest<T> {
+            val httpPayload = HttpService.Payload(url, method, params)
+            return HttpRequest(httpPayload, deserializer)
+        }
+
+        fun <T> makeRequest(
+            url: String,
+            method: String,
+            params: List<Pair<String, String>>,
+            body: String,
+            deserializer: DeserializationStrategy<T>
+        ): HttpRequest<T> {
+            val httpPayload = HttpService.Payload(url, method, params, body)
+            return HttpRequest(httpPayload, deserializer)
+        }
```

Comment:
I don't like all those methods, they add unnecesssary cluster. I think the thing that we need is to accept JSON object in Payload's secondary constructor. This way we can do:
```
return HttpRequest(Payload(url, "POST", body), InvokeFunctionResponse.serializer())
```